frontend/src/metabase/visualizations/echarts/cartesian/chart-measurements/index.ts

```diff
@@ -60,6 +60,37 @@ const getYAxisTicksWidth = (
   return Math.max(...measuredValues);
 };
 
+const getXAxisTicksWidth = (
+  dataset: ChartDataset,
+  settings: ComputedVisualizationSettings,
+  formatter: AxisFormatter,
+  renderingContext: RenderingContext,
+) => {
+  const xAxisDisplay = settings["graph.x_axis.axis_enabled"];
+
+  if (!xAxisDisplay) {
+    return 0;
+  }
+
+  if (xAxisDisplay === "rotate-90") {
+    return CHART_STYLE.axisTicks.size;
+  }
+
+  const tickWidths = dataset.map(datum => {
```

Comment:
I think it is more reasonable to check bordering ticks with since later in the calculation we assume they are at the ends of the axis

Reply:
Makes sense I can do that

Reply:
Done